Index: gdb-dashboard
===================================================================
--- gdb-dashboard	(リビジョン 1206)
+++ gdb-dashboard	(作業コピー)
@@ -811,6 +811,8 @@
         for number, line in enumerate(self.source_lines[start:end], start + 1):
             # properly handle UTF-8 source files
             line = to_string(line)
+            break_nums = self.find_breakpoint_locations(self.file_name)
+            is_breakpoint = str(number) in break_nums
             if int(number) == current_line:
                 # the current line has a different style without ANSI
                 if R.ansi:
@@ -823,6 +825,8 @@
                 else:
                     # just show a plain text indicator
                     line_format = number_format + '>{}'
+            elif is_breakpoint:
+                line_format = ansi(number_format, self.style_break) + ' {}'
             else:
                 line_format = ansi(number_format, R.style_low) + ' {}'
             out.append(line_format.format(number, line.rstrip('\n')))
@@ -835,9 +839,31 @@
                 'default': 5,
                 'type': int,
                 'check': check_ge_zero
+            },
+            'style_break': {
+                'default': '1;32;42'
             }
         }
 
+    def find_breakpoint_locations(self, filename):
+        output = run('info breakpoints')
+        output_lines = output.split('\n')
+
+        basename = os.path.basename(filename)
+
+        breakpoints = []
+
+        for line in output_lines:
+            # find string of type 'main.c:' located in the end of the line
+            filename_and_num_pos = line.find(basename + ':')
+            if filename_and_num_pos != -1:
+                loc = line[filename_and_num_pos:]
+                curr_filename, linenum = loc.split(':')
+
+                breakpoints.append(linenum)
+
+        return breakpoints
+
 class Assembly(Dashboard.Module):
     """Show the disassembled code surrounding the program counter. The
 instructions constituting the current statement are marked, if available."""
